### PR TITLE
added mule demo project to test the connector 

### PR DESCRIPTION
**Pull request by manoj-github-avio**

### CODE REVIEW

File: demo/azure-key-vault-connector-demo/pom.xml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            <artifactId>mule-apikit-module</artifactId>
+            <version>1.3.7</version>
+            <classifier>mule-plugin</classifier>
+        </dependency>
+        <dependency>
+        		<groupId>com.avioconsulting.mule.connector</groupId>
+			<artifactId>mule-azure-key-vault-connector</artifactId>
+			<version>1.0.0-SNAPSHOT</version>
+			<classifier>mule-plugin</classifier>
+        </dependency>
```
Comment (adesjardin) [inline demo/azure-key-vault-connector-demo/pom.xml:57]: We should move the version to a property to make it easy to update, and it should either be the latest released version (1.0.0) or the latest develop version (1.1.0-SNAPSHOT once @kkingavio updates the develop branch)

File: demo/README-demo.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Executing the Azure Key Vault Connector Demo
+
+
+## Step 2: Execute the Demo application
+
+Open the demo application (Azure-key-vault-connector-demo) with Anypoint Studio and start the application as a Mule Application
+
```
[the remark this answers is not in the document]
Comment (adesjardin) [reply demo/README-demo.md:7]: We probably should also make note that they should have done a mvn install on the connector first as well.

[the remark this answers is not in the document]
Comment (manoj-github-avio) [reply demo/README-demo.md:16]: added the above details in read me file

File: demo/azure-key-vault-connector-demo/pom.xml (hunk 1 above)
Comment (manoj-github-avio) [reply demo/azure-key-vault-connector-demo/pom.xml]: moved version of connector to property

**Pull request merged**